chevah/python-package
# [#3338] Use cryptography and pyOpenSSL 16.0.0 on ArchLinux builds

Dev environment does not work if openssl is built without SSLv3 support
=======================================================

The new builds for OpenSSL on ArchLinux are built without support for SSLv3 methods, this breaks the development environment set up because Python can't use urllib2.HTTPS* methods , as SSl.so is linked to a different OpenSSL (with support to SSLv3)

Further investigation showed that when building PyOpenSSL (0.13) or Python (2.7.10) against an OpenSSL with SSLv3 enabled it is impossible to disable those methods, so it would fail to run on systems with OpenSSL built with SSLv3 methods disabled.


Why we got into this (5 whys)
=============================

* Most distributions still build and distribute OpenSSL with SSLv3 support enabled.


Changes description
===================

* On ArchLinux build Python 2.7.10 with pyOpenSSL 16.0.0 with cryptography and cffi modules.


How to try and test the changes
===============================

reviewers @adiroiban 

* check that changes make sense
* build it on other OSes and check if ```./chevah_build test```runs ok. 
* build it on ArchLinux and check if ssl, pyOpenSSL and cryptography are imported

```
$ ./chevah_build build
$ ./chevah_build test
python 2.7.10 (default, Mar 29 2016, 16:29:32) 
[GCC 5.3.0]
zlib 1.0
stdlib ssl OpenSSL 1.0.2g  1 Mar 2016
cryptography 1.3.1 - OpenSSL OpenSSL 1.0.2g  1 Mar 2016
pyopenssl 16.0.0 - OpenSSL OpenSSL 1.0.2g  1 Mar 2016
PyCrypto 2.6.1
ctypes 1.1.0
setproctitle 1.1.8-dev0
```
